File: app/src/main/   java/com/gameboosterpro/app/MainActivity.java
package com.gamebooster.pro;

import android.app.Activity;
import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.provider.Settings;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends Activity {
    
    TextView statusText, ramText, pingText, rootText;
    Button gameModeBtn, bubbleBtn;
    
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        
        statusText = findViewById(R.id.statusText);
        ramText = findViewById(R.id.ramText);
        pingText = findViewById(R.id.pingText);
        rootText = findViewById(R.id.rootText);
        gameModeBtn = findViewById(R.id.gameModeBtn);
        bubbleBtn = findViewById(R.id.bubbleBtn);
        
        boolean root = RootUtil.isRooted();
        rootText.setText("ROOT: " + (root ? "YES - FULL POWER" : "NO - BASIC MODE"));
        
        startService(new Intent(this, BoosterService.class));
        
        gameModeBtn.setOnClickListener(v -> startGameMode());
        bubbleBtn.setOnClickListener(v -> showBubble());
    }
    
    void startGameMode() {
        try {
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                if (!Settings.System.canWrite(this)) {
                    Intent intent = new Intent(Settings.ACTION_MANAGE_WRITE_SETTINGS);
                    intent.setData(Uri.parse("package:" + getPackageName()));
                    startActivity(intent);
                    Toast.makeText(this, "Allow permission, then click again", Toast.LENGTH_LONG).show();
                    return;
                }
            }
            Settings.System.putInt(getContentResolver(), Settings.System.SCREEN_BRIGHTNESS, 255);
            
            if (RootUtil.isRooted()) {
                statusText.setText("Status: RED MAGIC MODE + ROOT BOOST");
            } else {
                statusText.setText("Status: GAME MODE ACTIVE");
            }
        } catch (Exception e) {
            statusText.setText("Status: ERROR");
        }
    }
    
    void showBubble() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            if (!Settings.canDrawOverlays(this)) {
                Intent intent = new Intent(Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
                        Uri.parse("package:" + getPackageName()));
                startActivity(intent);
                Toast.makeText(this, "Allow overlay, then click again", Toast.LENGTH_LONG).show();
                return;
            }
        }
        startService(new Intent(this, BubbleService.class));
    }
}

File: app/src/main/kotlin/com/gameboosterpro/app/MainActivity.kt
package com.gameboosterpro.app

import android.app.ActivityManager
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.provider.Settings
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var btnStartBubble: Button
    private lateinit var btnStopBubble: Button
    private lateinit var btnKillProcesses: Button
    private lateinit var tvStatus: TextView

    private val overlayPermissionLauncher = registerForActivityResult(
        ActivityResultContracts.StartActivityForResult()
    ) {
        if (Settings.canDrawOverlays(this)) {
            startBubbleService()
        } else {
            Toast.makeText(this, "Overlay permission denied", Toast.LENGTH_SHORT).show()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnStartBubble = findViewById(R.id.btnStartBubble)
        btnStopBubble = findViewById(R.id.btnStopBubble)
        btnKillProcesses = findViewById(R.id.btnKillProcesses)
        tvStatus = findViewById(R.id.tvStatus)

        btnStartBubble.setOnClickListener {
            handleStartBubble()
        }

        btnStopBubble.setOnClickListener {
            stopBubbleService()
        }

        btnKillProcesses.setOnClickListener {
            killBackgroundProcesses()
        }

        updateStatus()
    }

    override fun onResume() {
        super.onResume()
        updateStatus()
    }

    private fun handleStartBubble() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !Settings.canDrawOverlays(this)) {
            val intent = Intent(
                Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
                Uri.parse("package:$packageName")
            )
            overlayPermissionLauncher.launch(intent)
        } else {
            startBubbleService()
        }
    }

    private fun startBubbleService() {
        val intent = Intent(this, BubbleService::class.java)
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            startForegroundService(intent)
        } else {
            startService(intent)
        }
        Toast.makeText(this, "Bubble overlay started", Toast.LENGTH_SHORT).show()
        updateStatus()
    }

    private fun stopBubbleService() {
        val intent = Intent(this, BubbleService::class.java)
        stopService(intent)
        Toast.makeText(this, "Bubble overlay stopped", Toast.LENGTH_SHORT).show()
        updateStatus()
    }

    private fun killBackgroundProcesses() {
        val activityManager = getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
        val runningApps = activityManager.runningAppProcesses ?: return
        var killedCount = 0
        for (processInfo in runningApps) {
            if (processInfo.importance >= ActivityManager.RunningAppProcessInfo.IMPORTANCE_SERVICE
                && processInfo.processName != packageName
            ) {
                activityManager.killBackgroundProcesses(processInfo.processName)
                killedCount++
            }
        }
        Toast.makeText(this, "Killed $killedCount background process(es)", Toast.LENGTH_SHORT).show()
    }

    private fun updateStatus() {
        val running = BubbleService.isRunning
        tvStatus.text = if (running) "Bubble Overlay: ACTIVE" else "Bubble Overlay: INACTIVE"
        btnStartBubble.isEnabled = !running
        btnStopBubble.isEnabled = running
    }
}
